Code/Source/Entity/Terrain/TerrainEntityManagerSystemComponent.cpp

```diff
@@ -0,0 +1,154 @@
+        {
+            serializeContext->Class<TerrainEntityManagerSystemComponent, AZ::Component>()->Version(0);
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<TerrainEntityManagerSystemComponent>("Terrain Entity Manager", "Manages the RGL Terrain entity.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "RGL")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true);
+            }
+        }
+    }
+
+    void TerrainEntityManagerSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("TerrainEntityManagerService"));
+    }
+
+    void TerrainEntityManagerSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("TerrainEntityManagerService"));
+    }
+
+    void TerrainEntityManagerSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("RGLService"));
+    }
+
+    void TerrainEntityManagerSystemComponent::EnsureRGLEntityDestroyed()
+    {
+        if (m_rglEntity)
+        {
+            RGL_CHECK(rgl_entity_destroy(m_rglEntity));
+            m_rglEntity = nullptr;
+        }
+
+        if (m_rglMesh)
+        {
+            RGL_CHECK(rgl_mesh_destroy(m_rglMesh));
+            m_rglMesh = nullptr;
+        }
+    }
+
+    void TerrainEntityManagerSystemComponent::UpdateWorldBounds()
+    {
+        AZ::Aabb newWorldBounds = AZ::Aabb::CreateFromPoint(AZ::Vector3::CreateZero());
+        AzFramework::Terrain::TerrainDataRequestBus::BroadcastResult(
+            newWorldBounds, &AzFramework::Terrain::TerrainDataRequests::GetTerrainAabb);
+
+        if (!m_terrainData.UpdateBounds(newWorldBounds))
+        {
+            // The terrain data was not updated. No need to update the RGL mesh.
+            return;
+        }
+
+        EnsureRGLEntityDestroyed();
+
+        Utils::SafeRglMeshCreate(
+            m_rglMesh,
+            m_terrainData.m_vertices.data(),
+            m_terrainData.m_vertices.size(),
+            m_terrainData.m_indices.data(),
+            m_terrainData.m_indices.size());
+
+        if (!m_rglMesh)
+        {
+            AZ_Assert(false, "The TerrainEntityManager was unable to create an RGL mesh.");
+            return;
+        }
+
+        Utils::SafeRglEntityCreate(m_rglEntity, m_rglMesh);
+        if (!m_rglEntity)
+        {
+            AZ_Assert(false, "The TerrainEntityManager was unable to create an RGL entity.");
+            return;
+        }
+
+        RGL_CHECK(rgl_entity_set_pose(m_rglEntity, &Utils::IdentityTransform));
```

Comment:
I am not sure here - does terrain have any trasform in o3de? Technically it can have one - please double check.

Reply:
I will create an issue for that. The code is marked as added because it was moved and IMO this is outside the scope of this pr.